flocker_bb/builders/maint.py

```diff
@@ -25,6 +43,128 @@ def makeCleanOldBuildsFactory():
     return factory
 
 
+def get_creation_time(node):
+    """
+    Get the creation time of a libcloud node.
+
+    Rackspace and EC2 store the information in different metadta.
+
+    :return datetime:
+    """
+    date_string = node.extra.get("created", node.extra.get("launch_time"))
+    if date_string is None:
+        return None
+    else:
+        return parse_date(date_string)
+
+
+@attributes([
+    Attribute('lag'),
+    Attribute('prefixes',
+              default_value=('acceptance-test-', 'client-test-')),
+])
+class CleanAcceptanceInstances(LoggingBuildStep):
+    """
+    :ivar timedelta lag: THe age of instances to destroy.
+    :param prefixes: List of prefixes of instances to destroy.
+    """
+    name = 'clean-acceptance-instances'
+    description = ['Cleaning', 'acceptance', 'instances']
+    descriptionDone = ['Clean', 'acceptance', 'instances']
+    haltOnFailure = False
+    flunkOnFailure = True
+
+    def _thd_clean_nodes(self, config):
+        # Get the libcloud drivers corresponding to the acceptance tests.
+        config = yaml.safe_load(config)
+        rackspace = get_driver(Provider.RACKSPACE)(
+            config['rackspace']['username'], config['rackspace']['key'],
+            region=config['rackspace']['region'])
+        ec2 = get_driver(Provider.EC2)(
+            config['aws']['access_key'], config['aws']['secret_access_token'],
+            region=config['aws']['region'])
+
+        drivers = [rackspace, ec2]
+
+        # Get the prefixes of the instance names, appending the creator from
+        # the config.
+        creator = config['metadata']['creator']
+        prefixes = tuple(map(lambda prefix: prefix + creator, self.prefixes))
+
+        # Find out the cutoff time to use.
+        now = datetime.now(tzutc())
+        cutoff = now - self.lag
+
+        # Get all the nodes from the cloud providers
+        all_nodes = []
+        for driver in drivers:
+            all_nodes += driver.list_nodes()
+
+        # Filter them for running nodes with the right prefix.
+        test_nodes = [
+            node for node
+            in all_nodes
+            if node.name.startswith(prefixes)
+            and (node.state == NodeState.RUNNING)
```

Comment:
On Rackspace you still pay for Stopped nodes. Can't we just destroy them regardless of their state?